src/context/trader_risk_group.tsx

```diff
@@ -475,29 +457,91 @@ export const TraderProvider: FC<{ children: ReactNode }> = ({ children }) => {
     setTraderVolume('0')
   }
 
-  const refreshTrg = useCallback(async () => {
-    await setTRGDetails()
-  }, [setTRGDetails, rawData.trg])
-
-  const refreshMpg = useCallback(async () => {
-    await setMPGDetails()
-  }, [setMPGDetails, rawData.mpg])
+  const updateTRGDetails = (traderRiskGroup, rawData) => {
+    setTraderRiskGroup(traderRiskGroup)
+    if (traderRiskGroup) {
+      setRawData((prevState) => ({ ...prevState, trg: rawData }))
+    }
+  }
 
   useEffect(() => {
     perpsWasm()
   }, [rawData.mpg, rawData.trg])
 
   useEffect(() => {
-    const t2 = setInterval(refreshMpg, 2000)
-    return () => clearInterval(t2)
-  }, [currentMPG])
+    ;(async () => {
+      if (!currentMPG) return
+      const mpgData = await MarketProductGroup.fetch(connection, currentMPG)
+      updateMPGDetails(mpgData ? mpgData[0] : null, mpgData ? mpgData[1] : null)
+    })()
+  }, [currentMPG, connection])
 
   useEffect(() => {
-    if (wallet.connected) {
-      const t2 = setInterval(refreshTrg, 1000)
-      return () => clearInterval(t2)
-    } else {
-      setDefaults()
+    let id = null
+    let retryCount = 0
+
+    const subscribeMPG = async () => {
+      try {
+        if (currentMPG) {
+          id = connection.onAccountChange(currentMPG, async (info) => {
+            const trg = await MarketProductGroup.decode(info.data)
+            updateMPGDetails(trg, trg ? info : null)
+          })
+        } else {
+          setDefaults()
+        }
+      } catch (e) {
+        if (retryCount < MAX_RETRIES) {
+          retryCount++
+          subscribeMPG()
```

Comment:
should we by chance add a slight delay before retrying? It would retry in rapid succession and fail immediatly, perhaps a 500-1000ms delay?

Reply:
@deadeagle63 - i spoke with @candidatek and asked to update the implementation to use the hook. Best to use one convention across the app.

Regarding the use of a timeout on recurse, this is a good way to express:
```
        if (index > 5) return
	try {
		const response = await fetch(url);
		if (response.ok) {
			return response;
		} else {
			await new Promise(resolve => setTimeout(resolve, DELAY));
			return fetchWithRetries(url, index + 1);
		}
	} catch (error) {
		await new Promise(resolve => setTimeout(resolve, DELAY));
		return fetchWithRetries(url, index + 1);
	}
```